### PR TITLE
Feature/player overview

### PR DESCRIPTION
**Pull request by SecretJ12**
- Add overview for player
- Add links to player view and competition/tournament
- bugfixes

### CODE REVIEW

File: backend/src/main/java/de/secretj12/turnierplaner/db/repositories/MatchRepository.java [hunk 1]
```diff
@@ -30,4 +33,36 @@ public void deleteByComp(Competition competition) {
     public List<Match> nonGroupMatches(Competition competition) {
         return find("#nonGroupMatches", Parameters.with("comp", competition)).list();
     }
-}
+
+    public List<Match> filterMatches(Tournament tournament, Competition competition,
+                                     Player player,
+                                     Instant from, Instant to) {
+        StringBuilder query = new StringBuilder("FROM Match m WHERE TRUE");
```
Comment (FungOliver) [inline backend/src/main/java/de/secretj12/turnierplaner/db/repositories/MatchRepository.java:40]: Let's use [CriteriaQuery](https://docs.oracle.com/javaee/7/api/javax/persistence/criteria/CriteriaQuery.html) and not build Queries with a StringBuilder. I can add this this weekend. In Spring there would be `Specification` library, but I don't think there is a Quarkus equivalent, unfortunately.

Comment (SecretJ12) [reply backend/src/main/java/de/secretj12/turnierplaner/db/repositories/MatchRepository.java:40]: Fair point.
How about the quarkus variant: https://docs.quarkiverse.io/quarkus-jpastreamer/dev/index.html

Comment (FungOliver) [reply backend/src/main/java/de/secretj12/turnierplaner/db/repositories/MatchRepository.java:40]: Sure if you can build dynamic queries with it. I think CriteriaQuery is framework-independent, but if there is a framework-specific tool like the `Specification` API in Spring I would also prefer that.

Comment (FungOliver) [reply backend/src/main/java/de/secretj12/turnierplaner/db/repositories/MatchRepository.java:40]: I can also implement this if you want to this weekend. I just had to learn this last week ago anyway :)

Comment (SecretJ12) [reply backend/src/main/java/de/secretj12/turnierplaner/db/repositories/MatchRepository.java:40]: Works for dynamic queries, but doesn't really work out for this one.
Does not support matching on reference id.
So for checking the competition id we need to join competition and check for it's id afterwards. Ugly but works.
But we are not able to also join with tournament which makes it impossible to check tournament's id.

**Review by FungOliver — APPROVED** [review]

**Pull request merged**